src/navability/entities/Factor.jl

```diff
@@ -32,64 +32,49 @@ struct Factor
     _version::String
 end
 
-function InitializeFactorData()::FactorData
-    return FactorData(
-        false,
-        false,
-        [],
-        Dict{String,Any}(),
-        [],
-        [],
-        0.0,
-        0,
-        3.0
-    )
-end
+"""
+$(SIGNATURES)
+Create a prior for a Pose2 factor with a distribution Z representing (x,y,theta) prior information, 
+    e.g. `FullNormal([0.0, 0.0, 0.0], diagm([0.01, 0.01, 0.01]))`.
 
-function PriorPose2Data(;xythetaPrior = [0.0, 0.0, 0.0], xythetaCovars = [0.01, 0.01, 0.01])::FactorData
-    fnc = Dict(
-        "Z" => Dict(
-            "_type" => "IncrementalInference.PackedFullNormal",
-            "mu" => xythetaPrior,
-            "cov" => [xythetaCovars[1], 0.0, 0.0, 0.0, xythetaCovars[2], 0.0, 0.0, 0.0, xythetaCovars[3]],
-        )
-    )
-    data = InitializeFactorData()
-    data.fnc = fnc
-    data.certainhypo = [1]
+Default value of Z = `FullNormal([0.0, 0.0, 0.0], diagm([0.01, 0.01, 0.01]))`.
+"""
+function PriorPose2Data(;Z::Distribution = FullNormal([0.0, 0.0, 0.0], diagm([0.01, 0.01, 0.01])))::FactorData
```

Comment:
I don't like the default values and think they should be removed but I'm not going to do significant functional changes in this PR.

Reply:
Sorry my concern was more about the signature. Our distribution signature is accepted a "matrix". JS has no concept of matrix. We need to accept flat array or array of array.

Reply:
Ya we agreed to remove defaults for distributions. I'm okay with that going forward. Just need to update everything downstream when we do.

Reply:
I'm liking 'array of array' the more i work with json

Reply:
Array of array is what I have seen most but most arrays in JS are dense. This might be non ideal in some contexts. I think array of array is probably best option until we find use cases that break it.